From 9c72dafbe6443b61c5dda39b63edd261e9034989 Mon Sep 17 00:00:00 2001
From: HolonProduction <holonproduction@gmail.com>
Date: Sat, 31 Jan 2026 13:21:24 +0100
Subject: [PATCH] LSP: Add `godot` to known language ids
References: https://github.com/godotengine/godot/issues/115631
            https://github.com/godotengine/godot/pull/115671
Upstream: merged

fix communication between kate and godot via lsp

can be probably dropped if included in next release

---
 modules/gdscript/language_server/godot_lsp.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/modules/gdscript/language_server/godot_lsp.h b/modules/gdscript/language_server/godot_lsp.h
index 31a5bdc14320..f1401efa0b53 100644
--- a/modules/gdscript/language_server/godot_lsp.h
+++ b/modules/gdscript/language_server/godot_lsp.h
@@ -695,10 +695,12 @@ struct TextDocumentItem {
 		version = p_dict["version"];
 		text = p_dict["text"];
 
-		// Clients should use "gdscript" as language id, but we can't enforce it. The Rider integration
-		// in particular uses "gd" at the time of writing. We normalize the id to make it easier to work with.
+		// Clients should use "gdscript" as language id, but we can't enforce it.
+		// We normalize some known ids to make them easier to work with:
+		// Rider < 2026.1: "gd"
+		// Kate: "godot"
 		String rawLanguageId = p_dict["languageId"];
-		if (rawLanguageId == "gdscript" || rawLanguageId == "gd") {
+		if (rawLanguageId == "gdscript" || rawLanguageId == "gd" || rawLanguageId == "godot") {
 			languageId = LanguageId::GDSCRIPT;
 		} else {
 			languageId = LanguageId::OTHER;
